### PR TITLE
hmac test: fix set_client

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

Have to make sure to call the Digest set client.

### Testing Strategy

Trying to run the HMAC test.

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by alistair23 — APPROVED** [review]

**Review by twilfredo — APPROVED** [review]

**Review by lschuermann — APPROVED** [review]

File: capsules/extra/src/test/hmac_sha256.rs [hunk 1]
```diff
@@ -81,3 +82,8 @@ impl digest::ClientHash<32> for TestHmacSha256 {
         kernel::debug!("HMAC-SHA256 matches!");
     }
 }
+
+impl digest::ClientVerify<32> for TestHmacSha256 {
+    fn verification_done(&self, _result: Result<bool, ErrorCode>, _compare: &'static mut [u8; 32]) {
+    }
+}
```
Comment (brghena) [inline capsules/extra/src/test/hmac_sha256.rs:89]: Why did this empty `verification_done()` implementation get added? Is this related?

Comment (bradjc) [reply capsules/extra/src/test/hmac_sha256.rs:89]: Yes. We need this so that the `Digest` trait becomes implemented for `T` (`TestHmacSha256`), and therefore we can call the correct `set_client()` function.

In case you are wondering what happens to tests when they are not compiled & executed on any board...

Comment (brghena) [reply capsules/extra/src/test/hmac_sha256.rs:89]: Oh, I see. It was calling the incorrect `set_client` before.

**Review by brghena — APPROVED** [review]
